File: Project2/Model/Customer.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Project2.Model
{
    public class Customer
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public Guid CustomerId { get; set; }

        [Required]
        public string Name { get; set; }

        public string? Email { get; set; }

        public string? Phone { get; set; }

        public string? Address { get; set; }
        public ICollection<RentalDetail> RentalDetails { get; set; }
    }
}

File: Project2/ProjectUI/ProjectUI/app/model/Customer.js
Ext.define('ProjectUI.model.Customer', {
    extend: 'ProjectUI.model.Base',
    idProperty: 'CustomerId',
    fields: [
        { name: 'customerId', type: 'string' },
        { name: 'name', type: 'string' },
        { name: 'email', type: 'string' },
        { name: 'phone', type: 'string' },
        { name: 'address', type: 'string' }
    ]
});